### ISSUE TITLE
Request for Comments: the road to alpha

### ISSUE THREAD

**Issue by jnbooth** [naruto-unison/naruto-unison#46]
Alpha release was originally slated for January 1, 2020—just over two months from now. A few questions:

- In its current state, how close does Naruto Unison feel to being ready for alpha?
- What should be the project's priorities in preparing for alpha? This could be absolutely anything, including ideas that haven't shown up on the issues page; clans; UI; XP; more missions; even adding/reworking more characters.
- What steps should _not_ be prioritized until alpha, if any? E.g. maybe you think the remaining missions shouldn't be added until there is more opportunity for community feedback.

**Comment by Kerodas** [comment]
I think that the game is almost ready for alpha. At its' current stage all that is missing is exp system for both pvp and pve, ranks, ladder ranking and character trees for pve battles. Those should be the priority for now as they are core functions of the game that should keep players engaged.
Missions should not be added to alpha stage in my opinion. Alpha's purpose is public testing of how characters, matchmaking and balancing works. Missions would be better to keep till beta.
Progression should also be designed till alpha is released.
You could also try making 1 boss battle for alpha and see how it works out and as a reward for people who beat the boss during alpha stage, they could have something special like Naruto character with animated icons instead of usual icons.

**Comment by maxfire54** [comment]
**In its current state, how close does Naruto Unison feel to being ready for alpha?**
I think the game feels pretty ready-- we can't be prepared for every possible issue that may come up but I think we are quite prepared. I agree with statement above as well.

    **What should be the project's priorities in preparing for alpha? This could be absolutely anything, including ideas that haven't shown up on the issues page; clans; UI; XP; more missions; even adding/reworking more characters.**
I think the UI could use some design work, it looks a little sloppy and could be tightened up a little, there's a lot of open space-- at least in the battle screen

    **What steps should not be prioritized until alpha, if any? E.g. maybe you think the remaining missions shouldn't be added until there is more opportunity for community feedback.**
As for missions, I'm not sure. It'll be helpful to see the balance/difficulty of missions but at the same time it'd be nice to just try to see balance for the characters without having to unlock them all. Then again, progression is hard to measure without knowing how hard the missions are going to be.

**Comment by maxfire54** [comment]
A question for you @jnbooth how do you feel it's going and the timeline? You stated above it was slated for January 2020, is there any chance it might come earlier/later? How do YOU think progress is coming along and how the game is looking?

**Comment by jnbooth** [comment]
I do think aiming for some time during winter break would be convenient. As far as progress, UI is definitely the main factor holding it back at this stage. Naruto Unison is in dire need of people with experience in graphic design and adaptive web layout. The only image assets for the entire site are those goofy-looking scrolls on the character selection pane, and the only reason it can't be played on mobile is that I haven't figured out how to fit everything on a small screen. I don't really know how long it will take for me to bang my head against that wall before getting something that works.

Test suites are also important but coming along well . Progress has been a bit slow because I am _very_ busy this week. I am hopeful that if the team (current population: 1) gains more software developers, writing test suites will be their opportunity to get their feet wet and learn how the game engine works. Anyway, the XP system shouldn't take long to implement once I settle on how to structure it. Same goes for PvE, regarding which I agree that it would be best to implement a single demo boss and leave the rest until later.

> they could have something special like Naruto character with animated icons instead of usual icons

Neat idea! I will see about working something out along those lines. Another task that would benefit from someone with graphic design experience, I suppose.

> It'll be helpful to see the balance/difficulty of missions but at the same time it'd be nice to just try to see balance for the characters without having to unlock them all.

Could you elaborate on that? Mission progress can be seen from the game (by clicking on the "Mission" button on the character preview pane) and from the character's page in the guide (e.g. [localhost:3000/characters/naruto-uzumaki-(s)](http://localhost:3000/characters/naruto-uzumaki-(s)]. If you mean character balance in terms of win rate and pick rate, that can be viewed at [localhost:3000/admin/usage](http://localhost:3000/admin/usage).

**Comment by maxfire54** [comment]
How is progress coming along?

**Comment by jnbooth** [comment]
Right on track! The only significant commit left for this year is an update to matchmaking, which should be done soon, and then perhaps some more features for the forums.

I've also been toying around with a way to update the server without having any downtime. It's a very minor feature, but one that seems doable. The idea is to run a new server in parallel to the old server, ensure all new games are handled by the new server, and then shut down the old server when its ongoing games all end. But that's all behind-the-scenes work.
